### ISSUE TITLE
Support Testing Memoized Components

### ISSUE THREAD

**Issue by xveganxxxedgex** [davidmfoley/isolate-components#24]
I'm trying to test a component that gets exported as memoized, like:

```
const SomeMemoizedComponent = memo(() => <div />)
export default SomeMemoizedComponent;
```

When I try to test `isolateComponent(<SomeMemoizedComponent />);`, the test seems to time out and won't execute. I've been able to get around this by doing either:

```
export const SomeComponent = () => <div />
export default memo(SomeComponent);
```

Then testing `isolateComponent(<SomeComponent />);`, or wrapping the component in the actual test like:

```
const Renderer = () => <SomeMemoizedComponent />
isolateComponent(<Renderer />);
```

But it'd be much more convenient to be able to pass the memoized component directly if possible

**Comment by davidmfoley** [comment]
Hmmm I can't reproduce a difference between a named function wrapped with `memo` and an anonymous one. Can you share an example that times out?

Here's the test that I *think* covers this behavior:

https://github.com/davidmfoley/isolate-react/blob/main/isolate-react/test/isolateComponent/react_memo.test.tsx#L13

**Comment by xveganxxxedgex** [comment]
My apologies! It looks like I was running into this in two different situations. One doesn't seem to be a problem with this repo specifically and seems to be caused by using `useDisclosure` from the `chakra-ui/react` library, so that's a problem on my end.

But the other case looks to be components that use `memo(forwardRef(...))` together. If I remove either of those and just do `memo(...)` or `forwardRef(...)` alone, the tests will pass, but using them together returns an error `Failed to render <Item>: t.type is not a function`

**Comment by xveganxxxedgex** [comment]
It is interesting though, because for that first case using that `useDisclosure` hook, if I export and test the non-memoized component, the tests will work as expected. So something may be triggering the hook to execute unexpectedly and get caught in a loop when wrapped in the memo handler?

**Comment by davidmfoley** [comment]
Fixed in v2.4.1 on npm

**Comment by xveganxxxedgex** [comment]
Thanks so much for pushing this fix! Would you by chance expect `setState` calls to work with the latest updates? I may be seeing an issue where if I update the state from within the test, the isolate wrapper doesn't rerender the inner forward ref component. Here's a test case of what I mean (in `react_memo.test.tsx`):

```
it('memoed forwardRef', () => {
    let setValue: Function
    const InnerButton = () => {
      const [text, setText] = React.useState('Default')
      setValue = setText
      return <button>{text}</button>
    }
    const RefButton = React.forwardRef(InnerButton)
    const Memoed = React.memo(RefButton)
    const isolated = isolateComponent(<Memoed />)
    isolated.findOne('button')
    assert.strictEqual(isolated.findOne('button').content(), 'Default')
    setValue('Edited')
    assert.strictEqual(isolated.findOne('button').content(), 'Edited')
  })
```
